### ISSUE TITLE
num_priors mismatch when calculating loss

### ISSUE THREAD

**Issue by zzzkkkyyy** [qfgaohao/pytorch-ssd#49]
I've tried a simple train process on my Ubuntu 16.04 PC, but when calculating the loss, it seems variables (confidence, locations, labels, boxes) have different 'num_priors' and cannot calculate the loss. I've set 'target_transform' parameter on and used the VOC07 dataset.

The pytorch version is 0.4.1 and the python version is 3.5.2. (I have modified the 'pathlib' format to 'os.path' format since pathlib is supported only after 3.6)

The Pycharm debugging process's screenshot is below (basenet is vgg16 mentioned in the project):
![pytorch-ssd-question](https://user-images.githubusercontent.com/29513086/56706040-0756a000-6746-11e9-977f-57a3334449ef.png)
Thanks!

**Comment by zzzkkkyyy** [comment]
sorry for my carelessness, I forgot that I commented extra layers of the network before...
Please ignore this stupid issue, thanks anyway for your code!

P.S. 
One small correction that can be done when running in pytorch 0.4.1. When calculating smooth_l1_loss for predicted_locations and gt_locations, gt_locations should be casted to type 'float' first, otherwise an type mismatching error will appear.
